src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -543,13 +548,27 @@ uct_cuda_copy_md_open(uct_component_t *component, const char *md_name,
         return UCS_ERR_NO_MEMORY;
     }
 
-    md->super.ops              = &md_ops;
-    md->super.component        = &uct_cuda_copy_component;
-    md->config.alloc_whole_reg = config->alloc_whole_reg;
-    md->config.max_reg_ratio   = config->max_reg_ratio;
-    *md_p                      = (uct_md_h)md;
+    md->super.ops               = &md_ops;
+    md->super.component         = &uct_cuda_copy_component;
+    md->config.alloc_whole_reg  = config->alloc_whole_reg;
+    md->config.max_reg_ratio    = config->max_reg_ratio;
+    md->config.dmabuf_supported = uct_cuda_copy_md_is_dmabuf_supported();
+
+    if ((config->enable_dmabuf == UCS_YES) && !md->config.dmabuf_supported) {
+        ucs_error("dmabuf support requested but not found");
+        goto err_not_supported;
+    } else if (config->enable_dmabuf == UCS_NO) {
+        md->config.dmabuf_supported = 0;
+        ucs_debug("dmabuf support is disabled by configuration");
+    }
+
+    *md_p = (uct_md_h)md;
 
     return UCS_OK;
+
+err_not_supported:
+    ucs_free(md);
+    return UCS_ERR_UNSUPPORTED;
```

Comment:
jump here from line 548

Reply:
ok, but it will be more code:
- i'll have to introduce `status` variable
- i'll need to have 2 labels (just for exit and for freeing the md)

Reply:
it's more consistent with our style that the function will not have many exit points